app/lib/actions/katello/product/destroy.rb

```diff
@@ -99,6 +89,20 @@ def view_versions(product)
           end
           results.join(', ')
         end
+
+        def check_ready_to_delete(product, organization_destroy)
+          unless organization_destroy || product.user_deletable?
+            if product.redhat?
+              fail _("Cannot delete Red Hat product: %{product}") % { :product => product.name }
+            elsif !product.published_content_view_versions.not_ignorable.empty?
+              fail _("Cannot delete product with repositories published in a content view.  Product: %{product}, %{view_versions}") %
+                     { :product => product.name, :view_versions => view_versions(product) }
+            elsif product.repositories.any? { |repo| repo.filters.any? { |filter| filter.repositories.size == 1 } }
+              fail _("Cannot delete product: %{product} with repositories that are the last affected repository in content view filters. Delete these repositories before deleting product.") %
```

Comment:
Does it have to be this way? Can't we just default to deleting the filters like we do with individual repo deletes?

Reply:
This would be the better behavior in case someone wanted to retain their filters etc. This is also the behavior for repos in published CVs so it makes sense to me to follow the same user flow. Deleting filters would be a non-reversible action and affect content in CVs, probably something user should have to look at and decide per CV.

Reply:
ahh, gotcha! Ok, just fix the typos then and I'm good :)